avocado/core/mux.py

```diff
@@ -29,6 +36,152 @@
 REMOVE_VALUE = 1
 
 
+class MuxTree(object):
+
+    """
+    Object representing part of the tree from the root to leaves or another
+    multiplex domain. Recursively it creates multiplexed variants of the full
+    tree.
+    """
+
+    def __init__(self, root):
+        """
+        :param root: Root of this tree slice
+        """
+        self.pools = []
```

Comment:
Not sure I understand the `short commit` approach. If that means to extract this change and make it before moving the mux here than it'd be quite hard (not impossible, though). If it's just about saying this part is about to be removed because... than I don't like the idea.

Anyway would the extra note explaining the minor change of the inner API in the commit message be enough?